### ISSUE TITLE
Integrate analysis in the JSON file generation

### ISSUE THREAD

**Issue by aaronfay** [kiyanaw/bloomfield#12]
We have an API that can used for bulk lookup of "correctly spelled" words in the language, it is available here:

```
curl -XPOST -H'Content-type: application/json' \
  -d '["kîtahtawê", "iskotêw", "kî-osîhtamâsôwak,", "ômisi", "ê-tôtahkik"]' \
  https://api.kiyanaw.net/bulk-lookup
```

You need to provide a `POST` request and pass in a list of items to look up, the results will be a dictionary that has a key entry for each of the words that was passed in:

```
{
  "kîtahtawê": [
    "kîtahtawê+Ipc"
  ],
  "iskotêw": [
    "iskotêw+N+I+Sg"
  ],
  "kî-osîhtamâsôwak,": [],
  "ômisi": [
    "PV/o+PV/misi+_+Fut+Cond+Err/Frag",
    "PV/o+PV/misi+_+Cnj+Err/Frag",
    "PV/o+PV/misi+_+Ind+3+Err/Frag",
    "PV/o+PV/misi+_+Imp+Err/Frag"
  ],
  "ê-tôtahkik": [
    "PV/e+tôtam+V+TI+Cnj+3Pl"
  ],
  "_suggestions": {}
}
 
```

In the cases where there is a _matching_ key in the response dictionary, _check that there is at least one item in the array that does not have `Err/Frag`_, if so then grab all of the non-`Err/Frag` items in the list (or just delete the erroneous ones) and capture that for the analysis in the JSON (see `iskotêw` as an example below):

```
    "analysis": [
        {
            "surface": "kîtahtawê",
            "analysis": null
        },
        {
            "surface": "iskotêw",
            "analysis": ["iskotêw+N+I+Sg", "iskotêw+N+I+Sg"]
        },
```

In the case where there is an empty list (eg: `kîtahtawê` above), just use `null` for the analysis for now.

### Missing analysis actions

For any of the items that are missing analysis (all `Err/Frag` or no items), print those to a separate file for each of the stories for futher analysis.

**Comment by Nashtronaut** [comment]
Hey @aaronfay, I got a pretty good start and a plan rolling tonight. I just have a few questions. 

Do you want to process `child.attr.canon`? that was my assumption as we are returning
```js
return { surface: child.attr.canon, analysis: "TODO" }
```
currently. 

Another question, is that we are calling the analysis process on each child. That would mean we are either making a ton of API requests, or feeding singular words into those provided files, rather than one bulk call. Is that okay? Is that what was expected? Please note that this is just how it is currently structured. We could write something that processes the entirety of the `.json` files afterwards as well. Although, it would be nice to get it all done as we go rather than writing the files then rewriting them. Up to you! We can meet about these questions as well, might be easier if you've got the time. 

Below is what I mean, 

```js
const processAnalysis = (node) => {
  ...
  if (child.name === 'w') {
    // Process analysis here on single node, rather than a bulk request to api. 
    // ex. POST ... body: ['single word'];
  }
  ...
};
```

So we end up with a bunch of little requests instead of a sizable singular one.

**Comment by aaronfay** [comment]
> Do you want to process child.attr.canon? 
Yeah, I'm pretty sure that's the one. 

> We could write something that processes the entirety of the .json files afterwards as well.

I just found the [Node.js version](https://github.com/UAlbertaALTLab/hfst-optimized-lookup/tree/main/node) so we can probably do the analysis in the same process, something like: 

```javascript
const { Transducer } = require('hfstol');
const fst = new Transducer('crk-strict-analyzer.hfstol');
fst.bulk_lookup(listOfWords)
```

FSTs can be found here: https://github.com/giellalt/lang-crk/releases/tag/fst-v2021.7.8

**Edit**

For any of the results that come back with `Err/Frag` (or whatever) we'll want to keep track of those, maybe write them out to a file like `SSSC-01-errfrag.json` or something, we'll want to keep track of their location in the file, so the file might look kinda like:

```
{
  "p1": {
    "s1": ["word", "word", ...]
  }
}
```

We could also do CSV if that's just as easy.

**Comment by Nashtronaut** [comment]
And just to further clarify, sorry if this was already mentioned, but lets say we got

```
wordResponse {  
  return 1: not fragged,
  return 2: not fragged,
  return 3: fragged,
};
```

would this be considered a success because we have SOME pieces that are not fragged? I believe there can be multiple responses for the words right? 

edit: ping! @aaronfay

**Comment by aaronfay** [comment]
@Nashtronaut we want to parse them out, so anything with an analysis goes into the `analysis` field, and anything that has `Err/Frag` goes into the separate file


I gave an example of analysis above, but actually I want to adjust the structure:
```
    "analysis": {
        "iskotêw": ["iskotêw+N+I+Sg", "iskotêw+N+I+Sg"]
    },
```

That will make lookup easy, however if there is no analysis, don't put an entry in the `analysis` object.

**Comment by aaronfay** [comment]
To clarify further, that `["iskotêw+N+I+Sg", "iskotêw+N+I+Sg"]` above ☝🏻 will be the response from the fst lookup

**Comment by Nashtronaut** [comment]
Okay, sounds good!

**Comment by aaronfay** [comment]
Further clarification, if you get fragmented responses, just discard them, but any good responses we can keep:

```
["iskotêw+N+I+Sg", "iskotêw+N+I+Sg", "iskotêhk+Err/Frag"] << just ignore the last one, keep the first two
```

If all responses are `Err/Frag` then don't include that in `analysis`.

**Comment by Nashtronaut** [comment]
> Further clarification, if you get fragmented responses, just discard them, but any good responses we can keep:
> 
> ```
> ["iskotêw+N+I+Sg", "iskotêw+N+I+Sg", "iskotêhk+Err/Frag"] << just ignore the last one, keep the first two
> ```
> 
> If all responses are `Err/Frag` then don't include that in `analysis`.

I want to apologize. I reread the original issue and answered a lot of my own questions.

**Comment by Nashtronaut** [comment]
Closed as of #14
